packages/core/src/components/forms/_input-group.scss

```diff
@@ -37,38 +41,48 @@ $input-button-height: $pt-button-height - 6px !default;
 $input-button-height-large: $pt-button-height !default;
 
 .pt-input-group {
-  display: block;
-  position: relative;
+  // .pt-input-group should be styled to look like a .pt-input
+  @include pt-flex-parent();
+  border-radius: $pt-border-radius;
+  box-shadow: input-transition-shadow($input-shadow-color-focus), $pt-input-box-shadow;
+  background: $input-background-color;
+
+  &.pt-fill {
+    @include pt-flex-parent-fill();
+  }
 
-  .pt-input {
-    // explicit position prevents shadow clipping https://github.com/palantir/blueprint/pull/490#issuecomment-273342170
-    position: relative;
-    width: 100%;
+  // apply the :focus state on .pt-input-group, not .pt-input
+  &:focus,
```

Comment:
🙅‍♂️ unfortunately this is probably a non-starter. can't shift the `:focus` state to the parent element because
a) `.pt-input-group` not actually focusable (no tab index, etc), and
b) it can't be made to appear focused **in css only** when its child `input` is focused because CSS can only go down the tree, not up.

you'll note that InputGroups now have no focus state whatsoever when typing in them.
https://6591-71939872-gh.circle-artifacts.com/0/home/circleci/project/packages/docs-app/dist/index.html#core/components/forms/input-group
![screen shot 2017-12-14 at 1 24 50 pm](https://user-images.githubusercontent.com/464822/34015068-34a80298-e0d2-11e7-924b-9574150c9991.png)

Reply:
@giladgray stashing this and punting for now